### PR TITLE
Upgrade

### PR DESCRIPTION
**Pull request by RomainPetit1**
🐛 Bug fixes:
  - Handle json error
  - fix schedule not ending 
  - handle dublicates events

### CODE REVIEW

File: Sources/PlayerAnalytics.swift [hunk 1]
```diff
@@ -80,6 +83,7 @@ public class PlayerAnalytics {
       switch result {
       case .success(_):
         self.sendPing(payload: self.buildPingPayload()) { (res) in
+            self.cleanEventsStack()
```
Comment (olivier-lando) [inline Sources/PlayerAnalytics.swift]: already called in `sendPing()` no ?

File: Sources/PlayerAnalytics.swift [hunk 2]
```diff
@@ -96,6 +100,7 @@ public class PlayerAnalytics {
...
```
Comment (olivier-lando) [inline Sources/PlayerAnalytics.swift]: already called in `sendPing()` no ?

File: Sources/PlayerAnalytics.swift [hunk 3]
```diff
@@ -173,43 +180,55 @@ public class PlayerAnalytics {
...
-    let session = RequestsBuilder().buildUrlSession()
-    TasksExecutor.execute(session: session, request: request) { (data, error) in
-      if data != nil {
-        let json = try? JSONSerialization.jsonObject(with: data!) as? [String: AnyObject]
-        if let mySession = json!["session"] as? String {
-          if self.sessionId == nil {
-            self.sessionId = mySession
+      if(eventsStack.count > 0){
+          var request = RequestsBuilder().postClientUrlRequestBuilder(apiPath: options.videoInfo.pingUrl)
+          print(options.videoInfo.pingUrl)
```
Comment (olivier-lando) [inline Sources/PlayerAnalytics.swift]: I would remove this

**Review by olivier-lando — CHANGES_REQUESTED** [review]

File: Sources/TasksExecutor.swift [hunk 4]
```diff
@@ -7,7 +7,11 @@ public class TasksExecutor: TasksExecutorProtocol {
   ) {
 
     let task = session.dataTask(with: request) { (data, response, error) in
-      completion(data, error)
+        if(data?.count == 0){
+            completion(data, UrlError.jsonError("error 400"))
+        }else{
+            completion(data, error)
+        }
```
Comment (ThibaultBee) [inline Sources/TasksExecutor.swift:37]: If you want to detect a server error, you should have a look at https://developer.apple.com/forums/thread/114119 instead

**Pull request merged**